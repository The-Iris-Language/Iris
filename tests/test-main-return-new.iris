.. Tests returning a new object
class Animal () {
    public:
        int legs;
        int eyes;
        bool fur;
        
        string noise(int volume) {
            return "ruh roh raggy";
        }

        Animal newAnimal() {
            Animal a = new Animal();
            Olympus.print("animal");
            return a;
        }

    permit: 
        string color;
        bool alive;
}

class Dog of Animal () {
    public:
        string noise(int volume) {
            return "bark";
        }
}

class Temp () {
    public: 
        int x;
    .~*
        void func (Animal a) {
            a = new Dog();
        } *~.
}


class Main () {
    public:
       int univ main () {
            Animal a = new Animal();
            a.legs = 3;
            Olympus.print("Set a.legs to 3");
            Animal a1 = a.newAnimal();
            a1.legs = 10;
            Olympus.print(a1.noise(10));
            Temp temp = new Temp();
            ..temp.func(a1);
            ..Olympus.print(a1.noise(10));
            ..Olympus.print("Set a1.legs to 10");
            return 0;
       }
}