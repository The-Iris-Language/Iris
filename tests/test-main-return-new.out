Set a.legs to 3
bark
